lib/charms/alertmanager_k8s/v0/alertmanager.py

```diff
@@ -43,7 +42,25 @@ def restore(self, snapshot):
         self.data = snapshot["data"]
 
 
-class AlertmanagerConsumer(ConsumerBase):
+class AlertmanagerConsumerEvents(ObjectEvents):
+    """Event descriptor for events raised by `AlertmanagerConsumer`."""
+
+    cluster_changed = EventSource(ClusterChanged)
+
+
+class RelationManagerBase(Object):
```

Comment:
It is not clear to me why you need this class.

Reply:
Same is repeated for both provider and consumer. Reduce duplication and (re)establish contract.